### DISCUSSION TITLE
Graph API sending email problems

### DISCUSSION THREAD

**Discussion by iunir** [O365/python-o365#1087] [Q&A] [answered]
Hello,

Having a web app, I would like to be able to send email on behalf of users. 
Users of my app are only corporate. I prefer to use auth_flow_type=='credentials'.

I've managed to write a script. Authorisation part passes successfully and I'm receiving the token.
But send fails. 

If I'm using the user_id as resource:
```
Client Error: 400 Client Error: Bad Request for url: https://graph.microsoft.com/v1.0/{{THERE_IS_CORRECT_USER_ID}}/sendMail | Error Message: Resource not found for the segment 'sendMail'. 
```

If I'm using email:
```
Client Error: 404 Client Error: Not Found for url: https://graph.microsoft.com/v1.0/users/{{THERE_IS_CORRECT_EMAIL}}/sendMail | Error Message: The requested user '{{THERE_IS_CORRECT_EMAIL}}' is invalid.
```

The permissions are set:
<img width="889" alt="image" src="https://github.com/O365/python-o365/assets/38693523/8f16e136-3877-485a-b9be-a971c53ee573">

I've also tried to do same using official [Postman collection](https://www.postman.com/microsoftgraph/workspace/microsoft-graph/collection/455214-085f7047-1bec-4570-9ed0-3a7253be148c/fork). Using email as resource provides me same result. But with user_id I'm getting:
```

    "error": {
        "code": "MailboxNotEnabledForRESTAPI",
        "message": "The mailbox is either inactive, soft-deleted, or is hosted on-premise."
    }
}
```
Also, I was able to list the users through API (I've took user Id from there).

Asking of help. What can be the reason? 

**UPDATE**
I've found it's not using *users/* before user_id, while using email as a resource it adds it automatically. 
Now I have same: **The mailbox is either inactive, soft-deleted, or is hosted on-premise** from my Python.

So, how to fix that problem?

**Comment by alejcas** [comment]
I think it's because you need to set the permissions to "delegated" and not "application".

can you triy this?

**Reply by iunir** [reply]
Thanks a lot @alejcas for quick response.

I've tried to set all possible **Delegated** permissions:
![image](https://github.com/O365/python-o365/assets/38693523/49218405-e81d-4d3d-b636-1771b602e13d)

But still same error. BTW in my understanding **Application** permissions for **auth_flow_type="credentials"** and **Delegated** permissions for authorising **On behalf of a user**. Please correct me if I'm wrong.

**Comment by iunir** [comment, marked as answer]
Finally. After about 1 week of struggling. Office 365 license ([business type](https://www.microsoft.com/en-us/microsoft-365/business/compare-all-microsoft-365-business-products), bought and assigned by admin) helped.
I've tried with @outlook.com user, bought private license, but not worked. Funny fact, Microsoft promised to give me 30d free, but charged first second.
Also tried [Microsoft Entra ID P2](https://learn.microsoft.com/en-us/entra/id-governance/licensing-fundamentals#features-by-license) license. Which did not help as well.

**Reply by alejcas** [reply]
There seems to be a limitation on delegate permissions with personal accounts
